neqo-udp/src/lib.rs

```diff
@@ -52,22 +52,44 @@ pub fn send_inner(
     Ok(())
 }
 
+#[cfg(unix)]
+use std::os::fd::AsFd as SocketRef;
+#[cfg(windows)]
+use std::os::windows::io::AsSocket as SocketRef;
+
 pub fn recv_inner(
     local_address: &SocketAddr,
     state: &UdpSocketState,
-    socket: quinn_udp::UdpSockRef<'_>,
+    socket: impl SocketRef,
```

Comment:
When reading an empty datagram `recv_inner` attempts another receive from the socket. `state.recv` would previously consume the `quinn_udp::UdpSocketRef`. Passing an `impl SocketRef` here allows to construct a `quinn_udp::UdpSocketRef` before each call to `state.recv`.